File: tests/07_phases_ompss.dg/fortran/success_task_31.f90
! <testinfo>
! test_generator=config/mercurium-ompss
! </testinfo>
SUBROUTINE FOO(N)
    IMPLICIT NONE
    INTEGER, INTENT(IN) :: N
    INTEGER :: RES

    INTEGER :: F, X
    F(X) = X + N

    RES = -1

    !$OMP TASK SHARED(RES, N)
        RES = F(2)
    !$OMP END TASK
    !$OMP TASKWAIT

    IF(RES /= 5) STOP 1

END SUBROUTINE FOO

PROGRAM P
    IMPLICIT NONE

    INTEGER :: N
    N = 3
    CALL FOO(N)
END PROGRAM P

File: tests/07_phases_ompss.dg/cxx/success_task_31.cpp
/*
<testinfo>
test_generator=(config/mercurium-ompss "config/mercurium-ompss-2 openmp-compatibility")
</testinfo>
*/

#pragma omp task inout(*var) cost(*var)
void foo(int* var)
{
    var++;
}

int main()
{
    int a;
    foo(&a);

    #pragma omp task cost(0)
    {
    }

    #pragma omp taskwait
}
